### PR TITLE
chore: [CSI-1445] use 8-byte for chain id storage

### PR DESCRIPTION
**Pull request by ujjwal-cyph**
Fixes https://app.clickup.com/t/37308523/CSI-1445

### CODE REVIEW

**Review by Suraj-Tiwari — DISMISSED** [review]

File: src/bitcoinForks/index.ts [hunk 1]
```diff
@@ -268,7 +268,7 @@ export default class BitcoinWallet implements Partial<IWallet> {
       chainIndex +
       addressIndex +
       contractDummyPadding +
-      '00'
+      intToUintByte(0, 64)
```
Comment (ghost) [inline src/bitcoinForks/index.ts]: Won't this break backward compatibility? I would suggest doing something similar to this https://app.clickup.com/t/37308523/CSI-1260

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/bitcoinForks/index.ts (hunk 1 above)
Comment (ujjwal-cyph) [reply src/bitcoinForks/index.ts]: yes it will break. will need to bump SDK version.

Comment (ghost) [reply src/bitcoinForks/index.ts]: Bumping sdk won't be enough also need to use featureMap in communications and modify the code accordingly

Comment (ujjwal-cyph) [reply src/bitcoinForks/index.ts]: Changes added

File: src/near/index.ts [hunk 2]
```diff
@@ -179,7 +192,9 @@ export default class NearWallet implements IWallet {
         transactionFees +
         decimal +
         contractDummyPadding +
-        (addAccount ? intToUintByte(1, 8) : intToUintByte(0, 8))
+        (addAccount
+          ? intToUintByte(1, 64)
```
Comment (ujjwal-cyph) [inline src/near/index.ts]:
```suggestion
          ? intToUintByte(1, longChainId ? 64 : 8)
```

File: src/solana/index.ts [hunk 3]
```diff
@@ -62,7 +70,7 @@ export default class SolanaWallet implements IWallet {
     // do nothing
   }
 
-  async generateMetaData(gasFees: number) {
+  async generateMetaData(sdkVersion: string, gasFees: number) {
```
Comment (Suraj-Tiwari) [inline src/solana/index.ts]: updating first param will also break backward compatibility

**Review by Suraj-Tiwari — CHANGES_REQUESTED** [review]

File: src/bitcoinForks/index.ts (hunk 1 above)
Comment (Suraj-Tiwari) [reply src/bitcoinForks/index.ts]: @TejasvCypherock good catch

**Review by Suraj-Tiwari — APPROVED** [review]

**Review by Suraj-Tiwari — APPROVED** [review]

**Review by Suraj-Tiwari — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**